test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,242 @@
+        auto random_qualities = seqan3::test::generate_sequence<seqan3::phred42>(default_sequence_length, 0, seed);
+        format_stream << id << '\n'  // write id
+                      << random_sequence << '\n' // write sequence
+                      << '+' << '\n' // write qualities separator
+                      << random_qualities << '\n'; // write qualities
+    }
+
+    return stream_buffer.str();
+}
+
+// ============================================================================
+// save file on disc temporarily
+// ============================================================================
+
+auto create_fastq_file(size_t const entries_size)
+{
+    // create temporary file, automatically removed on destruction
+    seqan3::test::tmp_filename fastq_file{"format_fastq_benchmark_test_file.fastq"};
+    auto fastq_file_path = fastq_file.get_path();
+
+    // fill temporary file with a fastq file
+    std::ofstream ostream{fastq_file_path};
+    ostream << generate_fastq_file(entries_size);
+    ostream.close();
+    return fastq_file;
+}
+
+// ============================================================================
+// write 3-line entry to stream as often as possible
+// ============================================================================
+
+void fastq_write_to_stream_seqan3(benchmark::State & state)
+{
+    std::ostringstream ostream;
+    seqan3::sequence_file_output fout{ostream, seqan3::format_fastq{}};
+
+    std::string const id{"@name"};
+    auto seq = seqan3::test::generate_sequence<seqan3::dna5>(default_sequence_length, 0, default_seed);
+    auto qual = seqan3::test::generate_sequence<seqan3::phred42>(default_sequence_length, 0, default_seed);
+
+    for (auto _ : state)
+        fout.emplace_back(seq, id, qual);
+}
+
+// ============================================================================
+// seqan3 benchmark
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// read dummy fastq file from a stream
+// ----------------------------------------------------------------------------
+
+void fastq_read_from_stream_seqan3(benchmark::State & state)
+{
+    std::string file = generate_fastq_file(state.range(0));
+
+    seqan3::dna5_vector sequence{};
+    std::string id{};
+    std::vector<seqan3::phred42> quality{};
+
+    for (auto _ : state)
+    {
+        state.PauseTiming();
+        std::istringstream istream{file};
+        state.ResumeTiming();
+
+        seqan3::sequence_file_input fin{istream, seqan3::format_fastq{}};
+        for (auto && [read_sequence, read_id, read_quality] : fin)
+        {
+            sequence = std::move(read_sequence);
+            id = std::move(read_id);
+            quality = std::move(read_quality);
+        }
+
+        sequence.clear();
+        id.clear();
+        quality.clear();
```

Comment:
```suggestion
        seqan3::sequence_file_input fin{istream, seqan3::format_fastq{}};
        auto it = fin.begin();
        for (size_t i = 0; i < iterations_per_run; ++i)
            it++;
```
This is how the fasta benchmark does it and I think the comparison to seqan2 is fairer this way because you are doing an additional move and clear, although the record already contains the data (and is cleared). 
In a real use case you would not store the data separately but work directly on the record no?

Reply:
No I think in a real scenario you are storing the sequences in a container. Which means the important comparison would be to store the data in a container and clearly you would then move it here. So this would be closer to a real scenario. But maybe we are not interested in the stuff around (can be part of a macro benchmark) and only try to measure the actual operation necessary. So I make a final commit to adapt the benchmark to the fasta benchmark.